executor/src/witgen/global_constraints.rs

```diff
@@ -237,8 +238,8 @@ fn propagate_constraints<T: FieldElement>(
             }
         }
         Identity::Lookup(LookupIdentity { left, right, .. })
-        | Identity::Permutation(PermutationIdentity { left, right, .. }) => {
```

Comment:
Is this a bug fix? Why do you mention `permutation` in the docstring below?

Reply:
Yes, @georgwiese flagged that we should not look at permutations here, since they constrain more than range checking.

Reply:
I had added permutation to the docstring originally since we were looking at them, removed again now that we are not.

Reply:
Actually, looking at the code again, in this match arm there are two things happening:
1. If there is a constraint like `[a] in [b]`, and `b` has a range constraint, the same range constraint should apply to `a`. This is actually true for `[a] is [b]` as well!
2. If there is a constraint like `[a] in [b]` and `b` is a fixed column of the form `i % (2**k)`, this adds a range constraint. In this case, the constraint is removed, because all it does is enforce the range constraint.

I think in case 2, we could do the same for permutations, but removing the permutation would be a bug. So maybe the best solution is to include permutations again but set `remove = matches!(identity, Identity::Lookup(..) | Identity::PhantomLookup(..));`.

Sorry for the back and forth!